src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp

```diff
@@ -963,13 +963,147 @@ ov::runtime::Tensor generate(const
+    int b_dim = 255 / b_step + 1;
+    auto input_yuv = std::vector<uint8_t>(height * b_dim * width * 3 / 2);
+    for (int b = 0; b <= 255; b += b_step) {
+        for (size_t y = 0; y < height / 2; y++) {
+            for (size_t x = 0; x < width / 2; x++) {
+                int r = static_cast<int>(y) * 512 / static_cast<int>(height);
+                int g = static_cast<int>(x) * 512 / static_cast<int>(width);
+                // Can't use random y/u/v for testing as this can lead to invalid R/G/B values
+                int y_val = ((66 * r + 129 * g + 25 * b + 128) / 256) + 16;
+                int u_val = ((-38 * r - 74 * g + 112 * b + 128) / 256) + 128;
+                int v_val = ((112 * r - 94 * g + 18 * b + 128) / 256) + 128;
+
+                size_t b_offset = height * width * b / b_step * 3 / 2;
+                size_t u_index = b_offset + height * width + y * width / 2 + x;
+                size_t v_index = u_index + height * width / 4;
+                input_yuv[u_index] = u_val;
+                input_yuv[v_index] = v_val;
+                size_t y_index = b_offset + y * 2 * width + x * 2;
+                input_yuv[y_index] = y_val;
+                input_yuv[y_index + 1] = y_val;
+                input_yuv[y_index + width] = y_val;
+                input_yuv[y_index + width + 1] = y_val;
+            }
+        }
+    }
+    return input_yuv;
+}
+
+void generate_tensor(ov::Tensor& tensor) {
+    size_t full_height = tensor.get_shape()[1];
+    size_t full_width = tensor.get_shape()[2];
+    int b_dim = static_cast<int>(full_height * 2 / (3 * full_width));
+    ASSERT_GT(b_dim, 1) << "Image height is invalid for I420 Accuracy test";
+    ASSERT_EQ(255 % (b_dim - 1), 0) << "Image height is invalid for I420 Accuracy test";
+    int b_step = 255 / (b_dim - 1);
+    auto input_image = color_test_image(full_width, full_width, b_step);
+    auto data_ptr = static_cast<uint8_t*>(tensor.data());
+    for (size_t j = 0; j < input_image.size(); ++j) {
+        data_ptr[j] = input_image[j];
+    }
+}
+
+template<typename T>
+ov::runtime::Tensor generate(const
+                             std::shared_ptr<ov::op::v8::I420toRGB>& node,
+                             size_t port,
+                             const ov::element::Type& elemType,
+                             const ov::Shape& targetShape) {
+    ov::Tensor tensor(elemType, targetShape);
+    generate_tensor(tensor);
+    return tensor;
+}
+template<typename T>
+ov::runtime::Tensor generate(const
+                             std::shared_ptr<ov::op::v8::I420toBGR>& node,
+                             size_t port,
+                             const ov::element::Type& elemType,
+                             const ov::Shape& targetShape) {
+    ov::Tensor tensor(elemType, targetShape);
+    generate_tensor(tensor);
+    return tensor;
+}
+} // namespace convert_i420
+
+namespace convert_nv12 {
+inline std::vector<uint8_t> color_test_image(size_t height, size_t width, int b_step) {
```

Comment:
Can we reuse implementation (https://github.com/openvinotoolkit/openvino/blob/master/src/tests/functional/shared_test_classes/include/shared_test_classes/single_layer/convert_color_nv12.hpp#L61) instead of duplicating?